### PR TITLE
fix: should not return fields related to a deleted field

### LINKED ISSUE #6348: flask-rest-jsonapi returning fields related to a deleted field
**Describe the bug**
Say we want tickets under a ticket-tag, we use the endpoint `/ticket-tags/{ticket_tag_id}/tickets`, but if say that ticket_tag_id is deleted, flask-rest-jsonapi does not check for that, and returns the tickets (not deleted ones) under that ticket_tag_id.

**Expected behavior**
Flask-Rest-JsonAPI should not return fields related to a deleted field. 

**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6348 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
Check if it works with get_trashed

**Comment by prateekj117** [comment]
@iamareebjamal Made changes to accommodate condition of query parameter `get_trashed`.

**Comment by iamareebjamal** [comment]
Build is failing

File: app/api/helpers/db.py [hunk 1]
```diff
@@ -66,7 +67,11 @@ def safe_query(self, model, column_name, value, parameter_name):
     :return:
     """
     try:
-        record = self.session.query(model).filter(getattr(model, column_name) == value).one()
+        if request.args.get('get_trashed') == 'true':
+            record = self.session.query(model).filter(getattr(model, column_name) == value).one()
+        else:
+            record = self.session.query(model).filter(getattr(model, column_name) == value,
+                                                      model.deleted_at == None).one()
```
Comment (iamareebjamal) [inline app/api/helpers/db.py]: Duplication

File: app/api/helpers/db.py [hunk 2]
```diff
@@ -67,11 +67,11 @@ def safe_query(self, model, column_name, value, parameter_name):
...
     """
     try:
+        record = self.session.query(model).filter(getattr(model, column_name) == value)
         if request.args.get('get_trashed') == 'true':
-            record = self.session.query(model).filter(getattr(model, column_name) == value).one()
+            record = record.one()
         else:
-            record = self.session.query(model).filter(getattr(model, column_name) == value,
-                                                      model.deleted_at == None).one()
+            record = record.filter(model.deleted_at == None).one()
```
Comment (iamareebjamal) [inline app/api/helpers/db.py]: .one() is duplicated. Once you'll deduplicate it, 2 control flow blocks will also collapse to 1

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

File: tests/hook_main.py [hunk 3]
```diff
@@ -3593,6 +3593,7 @@ def custom_placeholder_sub_topic_get_detail(transaction):
         db.session.add(event_sub_topic)
 
         custom_placeholder = CustomPlaceholderFactory(event_sub_topic_id=1)
+        custom_placeholder.event_sub_topic_id = 1
```
Comment (iamareebjamal) [inline tests/hook_main.py]: Why?

Comment (prateekj117) [reply tests/hook_main.py]: @iamareebjamal Oh, sorry. This was for testing when I was not able to figure out the exact reason for tests failing. I will undo this.

**Comment by prateekj117** [comment]
@iamareebjamal Done. Please review.

**Comment by iamareebjamal** [comment]
Ensure no behaviour changes in frontend

**Comment by prateekj117** [comment]
@iamareebjamal Tested. Though I am not sure what unexpected changes can it bring in frontend. Let it be frontend or android, we wouldn't want to return fields related to a deleted field.

**Comment by iamareebjamal** [comment]
We do if it is admin viewing the deleted records. Again, you are not thinking about what can break, just what cannot

**Comment by prateekj117** [comment]
@iamareebjamal Tested. Nothing seems to break.

**Comment by iamareebjamal** [comment]
It not possible that nothing breaks. Show me screenshot of admin panel with deleted users

**Comment by prateekj117** [comment]
@iamareebjamal 

Screenshot from 2019-09-07 18-07-00

Screenshot from 2019-09-07 18-07-26

**Comment by iamareebjamal** [comment]
See hound error

**Comment by prateekj117** [comment]
@iamareebjamal Weirdly enough, using `is` instead of `==` in comparing to none inside of `filter` is not working as expected.

**Comment by iamareebjamal** [comment]
Yeah, this was reported once more.

Approving it, but will merge only after a week to ensure all bug fixes are accomodated for 1.6.0

**Pull request merged**